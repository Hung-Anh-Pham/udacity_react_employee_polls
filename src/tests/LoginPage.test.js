import { MemoryRouter } from "react-router-dom";
import LoginPage from "../components/LoginPage";
import { renderWithProviders } from "./setup-test";
import { fireEvent, screen } from "@testing-library/react";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();

describe("Test LoginPage component", () => {

    let store;

    beforeEach(() => {
        store = mockStore({
            users: {},
            questions: {},
            authedUser: null,
        });
    });

    it("Should correctly render the LoginPage component", () => {
        const { page } = renderWithProviders(
            <MemoryRouter>
                <LoginPage />
            </MemoryRouter>,
            { store }
        )

        expect(page).toMatchSnapshot();
    });

    it("Should log in successfully with a valid crendential", () => {
        const username = "sarahedo";
        const password = "password123";

        renderWithProviders(
            <MemoryRouter initialEntries={['/login']}>
                <LoginPage />
            </MemoryRouter>,
            { store }
        )

        const usernameInput = screen.getByPlaceholderText("Username");
        fireEvent.change(usernameInput, { target: { value: username } });

        const passwordInput = screen.getByPlaceholderText("Password");
        fireEvent.change(passwordInput, { target: { value: password } });

        const submitBtn = screen.getByRole("button", { name: /Login/i });
        expect(submitBtn).toBeInTheDocument();

        fireEvent.click(submitBtn);
    })
});