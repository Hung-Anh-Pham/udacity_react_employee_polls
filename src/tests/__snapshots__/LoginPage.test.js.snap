// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoginPage component Should correctly render the LoginPage component 1`] = `undefined`;
